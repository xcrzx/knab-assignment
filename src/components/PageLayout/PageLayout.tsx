import { Container, makeStyles } from "@material-ui/core";
import React from "react";
import { Footer } from "./components/Footer";
import { Header } from "./components/Header";

type PageLayoutProps = {
  title: string;
  children: React.ReactNode;
};

const useStyles = makeStyles(({ spacing }) => ({
  layout: {
    minHeight: "100vh",
    display: "grid",
    gridTemplateRows: "auto 1fr auto",
  },
  content: {
    paddingBottom: spacing(2),
    paddingTop: spacing(2),
  },
}));

export const PageLayout = ({ title, children }: PageLayoutProps) => {
  const classes = useStyles();

  return (
    <div className={classes.layout}>
      <Header title={title} />
      <Container component="main" className={classes.content}>
        {children}
      </Container>
      <Footer />
    </div>
  );
};
