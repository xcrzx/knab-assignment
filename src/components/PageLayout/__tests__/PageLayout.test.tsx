import { PageLayout } from "../PageLayout";
import { render } from "@testing-library/react";

describe("PageLayout Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <PageLayout title="Test">Page content</PageLayout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
