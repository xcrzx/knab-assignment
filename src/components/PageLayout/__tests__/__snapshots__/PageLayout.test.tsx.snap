// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-layout-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Test
        </h6>
      </div>
    </header>
    <main
      class="MuiContainer-root makeStyles-content-2 MuiContainer-maxWidthLg"
    >
      Page content
    </main>
    <footer
      class="MuiContainer-root makeStyles-footer-3 MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Copyright Â© 2021 Get Quote Inc. All Rights Reserved
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
